File: src/main/java/block/TileState.java
package block;

import gl.UVPair;

public class TileState {
    public final Block block;

    public TileState(Block block) {
        this.block = block;
    }

    public UVPair getUVForFace(EnumDirection face) {
       return  block.getUVForFace(face);
    }

    public boolean shouldRender() {
        return block.shouldRender();
    }
    public boolean isOpaque() {
        return block.isOpaque();
    }

    public EnumRenderLayer renderLayer() {
         return block.renderLayer();
    }

    public float getEmittance() {
        return block.getEmittance();
    }

    @Override
    public boolean equals(Object other) {
        if(other instanceof TileState) {
            return this.block == ((TileState)other).block;
        }
        return false;
    }
}

File: src/main/kotlin/block/TileState.kt
package block

import gl.UVPair

class TileState(var block: Block) {
    fun getUVForFace(face: EnumDirection): UVPair = block.getUVForFace(face)
    fun shouldRender(): Boolean = block.shouldRender()
    fun isOpaque(): Boolean = block.isOpaque()
    fun renderLayer(): RenderType = block.renderLayer()
    override fun equals(other: Any?): Boolean {
        if(other is TileState) {
            return this.block == other.block
        }
        return false
    }

    fun getEmittance(): Float = block.getEmittance()
}